### DISCUSSION TITLE
Option to disable waterlogging on turtles?

### DISCUSSION THREAD

**Discussion by Justinjah91** [cc-tweaked/CC-Tweaked#1151] [General]
I like that we have the ability to waterlog turtles now, but I would also love to have an option to disable it. Preferably, this would be done via the settings API so it could be done per turtle (or even per program), but I have no idea if that would be possible. 

My mining program is set up to block off holes in the tunnel so that mobs cant get into the mine from natural caverns and stuff. This includes blocking off water and lava sources. But when the turtle hits a big pocket of water like an aquifer, the sheer number of source blocks present means that the turtle itself becomes waterlogged, generating a new source block INSIDE my mining tunnel. This subsequently washes away all of the torches, allowing mobs to spawn (which kinda defeats the purpose).

Thoughts?

**Comment by piprett** [comment]
Issue: #1152 

Personally I think this should be added. I defenetly see the irritation of turtle getting waterlogged.

**Comment by Kan18** [comment]
What about "thick turtles" which fully fill the block? This would prevent breaking the laws of physics. (Maybe Plethora could add this after adding tiny turtles https://github.com/SquidDev-CC/plethora/issues/80 )

**Reply by Justinjah91** [reply]
Hey I'd go for that

**Reply by Lupus590** [reply]
If plethora ever gets updated. It would probably be better to suggest this to another peripheral dev who targets the current versions of MC that CC targets.
